sys/linux/devio.txt

```diff
@@ -0,0 +1,222 @@
+ioctl$USBDEVFS_SETCONFIGURATION(fd fd_usbfs, cmd const[USBDEVFS_SETCONFIGURATION], arg ptr[in, int32])
+ioctl$USBDEVFS_SUBMITURB(fd fd_usbfs, cmd const[USBDEVFS_SUBMITURB], arg ptr[in, usbdevfs_urb])
+
+# TODO: arg is a pointer to a previously submitted URB.
+ioctl$USBDEVFS_DISCARDURB(fd fd_usbfs, cmd const[USBDEVFS_DISCARDURB], arg ptr[in, int64])
+
+ioctl$USBDEVFS_DISCSIGNAL(fd fd_usbfs, cmd const[USBDEVFS_DISCSIGNAL], arg ptr[in, usbdevfs_disconnectsignal])
+ioctl$USBDEVFS_CLAIMINTERFACE(fd fd_usbfs, cmd const[USBDEVFS_CLAIMINTERFACE], arg ptr[in, int32])
+ioctl$USBDEVFS_RELEASEINTERFACE(fd fd_usbfs, cmd const[USBDEVFS_RELEASEINTERFACE], arg ptr[in, int32])
+ioctl$USBDEVFS_IOCTL(fd fd_usbfs, cmd const[USBDEVFS_IOCTL], arg ptr[in, usbdevfs_ioctl])
+ioctl$USBDEVFS_CLAIM_PORT(fd fd_usbfs, cmd const[USBDEVFS_CLAIM_PORT], arg ptr[in, int32])
+ioctl$USBDEVFS_RELEASE_PORT(fd fd_usbfs, cmd const[USBDEVFS_RELEASE_PORT], arg ptr[in, int32])
+ioctl$USBDEVFS_GET_CAPABILITIES(fd fd_usbfs, cmd const[USBDEVFS_GET_CAPABILITIES], arg ptr[out, int32])
+ioctl$USBDEVFS_DISCONNECT_CLAIM(fd fd_usbfs, cmd const[USBDEVFS_DISCONNECT_CLAIM], arg ptr[in, usbdevfs_disconnect_claim])
+ioctl$USBDEVFS_FREE_STREAMS(fd fd_usbfs, cmd const[USBDEVFS_FREE_STREAMS], arg ptr[in, usbdevfs_streams])
+ioctl$USBDEVFS_DROP_PRIVILEGES(fd fd_usbfs, cmd const[USBDEVFS_DROP_PRIVILEGES], arg ptr[in, int32])
+ioctl$USBDEVFS_GET_SPEED(fd fd_usbfs, cmd const[USBDEVFS_GET_SPEED])
+
+usbdevfs_ep {
+	num	int8:7[0:15]
+	dir	int8:1
+}
+
+usbdevfs_ctrltransfer {
+	bRequestType	flags[usb_request_type_flags, int8]
+	bRequest	flags[usb_requests, int8]
+	wValue		int16
+	wIndex		int16
+	wLength		len[data, int16]
+	timeout		int32
+	data		ptr[inout, array[int8]]
+}
+
+usb_request_type_flags = USB_DIR_OUT, USB_DIR_IN, USB_TYPE_MASK, USB_TYPE_STANDARD, USB_TYPE_CLASS, USB_TYPE_VENDOR, USB_TYPE_RESERVED, USB_RECIP_MASK, USB_RECIP_DEVICE, USB_RECIP_INTERFACE, USB_RECIP_ENDPOINT, USB_RECIP_OTHER, USB_RECIP_PORT, USB_RECIP_RPIPE
+
+usbdevfs_bulktransfer {
+	ep	usbdevfs_ep
+	pad0	int8
+	pad1	int16
+	len	len[data, int32]
+	timeout	int32
+	data	ptr[inout, array[int8]]
+}
+
+usbdevfs_getdriver {
+	interface	int32
+	driver		array[int8, MAXDRIVERNAME]
+}
+
+usbdevfs_connectinfo {
+	devnum	int32
+	slow	int8
+}
+
+usbdevfs_setinterface {
+	interface	int32
+	altsetting	int32
+}
+
+usbdevfs_urb [
+	urb_type_control	usbdevfs_urb_control
+	urb_type_bulk		usbdevfs_urb_bulk
+	urb_type_interrupt	usbdevfs_urb_interrupt
+	urb_type_iso		usbdevfs_urb_iso
+] [varlen]
+
+usbdevfs_urb_control {
+	type		const[USBDEVFS_URB_TYPE_CONTROL, int8]
+	endpoint	usbdevfs_ep
+	status		int32
+	flags		flags[urb_flags, int32]
+	buffer		ptr[in, usb_ctrlrequest]
+	buffer_length	len[buffer, int32]
+	actual_length	int32
+	start_frame	int32
+	u		const[0, int32]
+	error_count	int32
+	signr		int32
+	usercontext	ptr[in, array[int8]]
+	iso_frame_desc	const[0, intptr]
```

Comment:
Just remove this field. It's declared as an array of size 0, which means that it is essentially empty in this case.